checks/cs50/2017/x/scratch/checks.py

```diff
@@ -0,0 +1,155 @@
+import json
+import os
+import sys
+
+sys.path.append(os.getcwd())
+from check50 import *
+
+class Scratch(Checks):
+    
+    @check()
+    def valid(self):
+        """project exists and is valid Scratch program"""
+        self.exists("project.sb2")
+
+        # Ensure that unzipped .sb2 file contains .json file
+        self.spawn("unzip project.sb2").exit()
+        self.exists("project.json")
+    
+    @check("valid")
+    def two_sprites(self):
+        """project contains at least two sprites"""
+        project = json.loads(File("project.json").read())
+        num_sprites = 0
+
+        # Loop over children: includes sprites and stages.
+        for child in project["children"]:
+            # Only sprites have costumes.
+            if "costumes" in child:
+                num_sprites += 1
+
+        if num_sprites < 2:
+            raise Error("Only {} sprite{} found, 2 required.".format(num_sprites,
+                "" if num_sprites == 1 else "s"))
+
+    @check("valid")
+    def non_cat(self):
+        """project contains a non-cat sprite"""
+        project = json.loads(File("project.json").read())
+        non_cat = False
+    
+        # As a heuristic, only the cat has a "meow" sound.
+        for child in project["children"]:
+        
+            # Skip over any non-sprites (e.g. backdrops).
+            if "costumes" not in child:
+                continue
+            
+            # If sprite has no sounds, it's probably not a cat.
+            can_meow = False
+            if "sound" not in child:
+                return
+
+            for sound in child["sounds"]:
+                if sound["soundName"] == "meow":
+                    can_meow = True
```

Comment:
`can_meow = any(sound["soundName"] == "meow" for sound in child["sounds"])` is a sexier/slightly more efficient way to do this